expr/func_groupbynodes.go

```diff
@@ -0,0 +1,102 @@
+package expr
+
+import (
+	"github.com/grafana/metrictank/api/models"
+	"github.com/grafana/metrictank/errors"
+	"github.com/grafana/metrictank/schema"
+)
+
+type FuncGroupByNodes struct {
+	in         GraphiteFunc
+	aggregator string
+	nodes      []expr
+	byNode     bool
+}
+
+func NewGroupByNodesConstructor(groupByNode bool) func() GraphiteFunc {
+	return func() GraphiteFunc {
+		return &FuncGroupByNodes{byNode: groupByNode}
+	}
+}
+
+func (s *FuncGroupByNodes) Signature() ([]Arg, []Arg) {
+	// This function supports both groupByNode and groupByNodes,
+    // but the signatures of them are different in Graphite.
+    if !s.byNode {
+		return []Arg{
+			ArgSeriesList{val: &s.in},
+			ArgString{val: &s.aggregator, validator: []Validator{IsAggFunc}},
+			ArgStringsOrInts{val: &s.nodes},
+		}, []Arg{ArgSeries{}}
+	}
+    // groupByNode accepts either an integer node num or a string tag key
+	s.nodes = append(s.nodes, expr{})
+	return []Arg{
+		ArgSeriesList{val: &s.in},
+		ArgStringOrInt{val: &s.nodes[0]},
+		ArgString{val: &s.aggregator, validator: []Validator{IsAggFunc}},
+	}, []Arg{ArgSeries{}}
+}
+
+func (s *FuncGroupByNodes) Context(context Context) Context {
+	context.PNGroup = 0
+	return context
+}
+
+func (s *FuncGroupByNodes) Exec(dataMap DataMap) ([]models.Series, error) {
+	series, err := s.in.Exec(dataMap)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(series) == 0 {
+		return series, nil
+	}
+
+	if len(s.nodes) == 0 {
+		return nil, errors.NewBadRequest("No nodes specified")
```

Comment:
I don't think graphite errors out in this case, I think it just aggregates them all and leaves the target name empty

Reply:
I tried to test this on Grafana. If I don't specify any node (name, tags, etc.), Grafana didn't give any error, and it shows the name as "Value". so I think you are right.

But this is checked here:
https://github.com/grafana/metrictank/blob/master/expr/func_groupbytags.go#L45

Reply:
That's because groupByTags *does* throw on that case: https://github.com/graphite-project/graphite-web/blob/master/webapp/graphite/render/functions.py#L5460

graphite has eccentricities, but we don't want to start failing dashboards that previously worked.

Reply:
got it, make sense. I have removed this check.